cloudigrade/api/clouds/azure/models.py

```diff
@@ -238,3 +243,110 @@ def __repr__(self):
             f"updated_at=parse({updated_at})"
             f")"
         )
+
+
+@receiver(post_delete, sender=AzureCloudAccount)
```

Comment:
Instead of using AzureCloudAccount's post-delete signal, what would you think about having this be called from the `AzureCloudAccount.disable` method? We should already call the cloud-specific disable method from `CloudAccount.disable`, and that is already called from CloudAccount's pre-delete signal.

This raises the question of whether pre- or post-delete is the better place for this, but until I can remember the exact minutia for why AWS's implementation uses post-delete, I'm inclined to favor consistency between the two.

Thoughts?

Reply:
As I think about this, hooking into `disable` means we _probably_ get source's pause events working here, but that handling source's resume events could be a problem since we wouldn't have a way to recreate the customer's registration. 🤔

Reply:
Yes, that would be problematic,  deleting the registration assignment upon a pause means we won't be able to unpause when we get those events from sources.